Should skip already downloaded logs

```
$> ( source $PWD.secrets ; datalad run -m "A sample run of tinuous" tools/tinuous/tinuous -c tinuous-config.yaml; )
travis not found
[INFO   ] == Command start (output follows) =====
2021-02-02T17:42:48-0500 [INFO    ] tinuous Fetching logs from github
2021-02-02T17:42:48-0500 [INFO    ] tinuous Fetching runs newer than 2021-01-20 00:00:00+00:00
2021-02-02T17:42:49-0500 [INFO    ] tinuous Fetching runs for workflow test_crippled.yml (CrippledFS)
2021-02-02T17:42:49-0500 [INFO    ] tinuous Found run 1310
2021-02-02T17:42:49-0500 [INFO    ] tinuous Downloading logs for test_crippled.yml (CrippledFS) #1310 to 2021/02/02/github/pr/UNK/903a5c477420304228920f04f1db8300ec044879/CrippledFS/1310
Traceback (most recent call last):
  File "tools/tinuous/tinuous", line 491, in <module>
    main()
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "tools/tinuous/tinuous", line 474, in main
    bl.download(path_template)
  File "tools/tinuous/tinuous", line 230, in download
    zf.extractall(p)
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/zipfile.py", line 1647, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "/home/yoh/miniconda3/envs/tinuous/lib/python3.8/zipfile.py", line 1701, in _extract_member
    open(targetpath, "wb") as target:
PermissionError: [Errno 13] Permission denied: '2021/02/02/github/pr/UNK/903a5c477420304228920f04f1db8300ec044879/CrippledFS/1310/test/6_Install dependencies.txt'
[INFO   ] == Command exit (modification check follows) =====
[INFO   ] The command had a non-zero exit code. If this is expected, you can save the changes with 'datalad save -d . -r -F .git/COMMIT_EDITMSG'
CommandError: 'tools/tinuous/tinuous -c tinuous-config.yaml' failed with exitcode 1 under /home/yoh/proj/datalad/ci/logs
```

that file is under annex, so not writeable, but it seems to be "complete"
```
(git)smaug:~datalad/ci/logs[master]git
$> tail 2021/02/02/github/pr/UNK/903a5c477420304228920f04f1db8300ec044879/CrippledFS/1310/test/6_Install\ dependencies.txt
2021-02-02T20:53:49.3768024Z Using legacy 'setup.py install' for pyperclip, since package 'wheel' is not installed.
2021-02-02T20:53:49.3769605Z Using legacy 'setup.py install' for python-xmp-toolkit, since package 'wheel' is not installed.
2021-02-02T20:53:49.3771637Z Using legacy 'setup.py install' for requests-ftp, since package 'wheel' is not installed.
2021-02-02T20:53:49.3772712Z Building wheels for collected packages: datalad
2021-02-02T20:53:49.3775129Z   Building wheel for datalad (PEP 517): started
2021-02-02T20:53:50.3820687Z   Building wheel for datalad (PEP 517): finished with status 'done'
2021-02-02T20:53:50.3898520Z   Created wheel for datalad: filename=datalad-0.14.0-py3-none-any.whl size=1594462 sha256=0b62055cc01a62cdd017c4ff526011a0577b3dfb7eb9082ebe0025278f96df6e
2021-02-02T20:53:50.3901165Z   Stored in directory: /tmp/pip-ephem-wheel-cache-u741rwhi/wheels/95/2a/99/f2ee0f42b3c46f75a962d71ae4b3a4281214039391ad3ea460
2021-02-02T20:53:50.3952404Z Successfully built datalad
2021-02-02T20:53:50.4194822Z ##[error]The operation was canceled.
```

so I think it should not be "refetched"/"rewritten", or there is a reason for that?

--- a/tinuous
+++ b/tinuous
@@ -217,6 +217,15 @@ class GHABuildLog(BuildLog):
     def download(self, path_template: str) -> None:
         p = Path(path_template.format_map(self.path_fields()))
         p.mkdir(parents=True, exist_ok=True)
+        if any(p.iterdir()):
+            log.info(
+                "Logs for %s (%s) #%s already downloaded to %s; skipping",
+                self.workflow_file,
+                self.workflow_name,
+                self.number,
+                p,
+            )
+            return
         log.info(
             "Downloading logs for %s (%s) #%s to %s",
             self.workflow_file,
@@ -348,6 +357,14 @@ class TravisJobLog(BuildLog):
 
     def download(self, path_template: str) -> None:
         p = Path(path_template.format_map(self.path_fields()))
+        if p.exists():
+            log.info(
+                "Logs for job %s.%s already downloaded to %s; skipping",
+                self.number,
+                self.job,
+                p,
+            )
+            return
         p.parent.mkdir(parents=True, exist_ok=True)
         log.info(
             "Downloading logs for job %s.%s to %s",

